st2common/st2common/util/loader.py

```diff
@@ -176,9 +178,10 @@ def register_runner(module_name):
 
     LOG.debug('Loading runner from: %s', module_path)
 
-    module = imp.load_source(module_name, module_path)
+    if module_name not in RUNNER_MODULES:
```

Comment:
Yeah, I'm personally fine with caching an existing module. It's also worth noting that this requires process restart (action runner) in case module changes and user wants those changes to take an effect, but imo that's fine and just something we need to document.

One thing we could probably to avoid the need of restart (well in most cases) or avoid the need to cache is to use `.reload()` (or delete module from internal module cache), but that's not Erlang and it's not really that robust and good idea to use it in most cases so process restart is more robust.

Anyway, just something to document in pluggable runner section of documentation :P

/cc @bigmstone